src/constructs/autoscaling/asg.ts

```diff
@@ -30,7 +31,7 @@ export interface GuAutoScalingGroupProps
     AppIdentity,
     GuAsgCapacity {
   imageId?: GuAmiParameter;
-  imageRecipe?: string;
+  imageRecipe?: string | AmigoProps;
```

Comment:
would that not be a breaking change though??

Reply:
It would, yes. We'd need to ensure the [commit describes this fact](https://www.conventionalcommits.org/en/v1.0.0/) so that the release automation creates a new major version.

Reply:
ok but, why not keep it a non-breaking change. it still all relates to the recipe, and so expressing as a string (standard use case) is more succinct than forcing everyone to express it in object form when only specifying the recipe property.
if there's a convention in this repo to avoid what I've done here, more than happy to change it, but otherwise I'd prefer to keep as is

Reply:
We don't have a strong convention here. We should do though, as it'll create a more intuitive API. This isn't blocking, so happy to get this in and refactor later if necessary.

Reply:
AWS have [well reasoned opinions here](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#flat), which we should consider adopting across GuCDK (cc @guardian/devx-operations)